klasses/views.py

```diff
@@ -16,9 +18,19 @@ class BootcampViewSet(ListModelMixin, GenericViewSet):
 
     def get_queryset(self):
         """Make a queryset which optionally shows what runs are available for enrollment"""
-        queryset = BootcampRun.objects.all().select_related("bootcamp")
+        user = self.request.user
+        complete_applications = BootcampApplication.objects.filter(
+            user=user, state=AppStates.COMPLETE.value
+        )
+        # if user is an alumni or have already bought a course, show them all the bootcamps
+        if user.profile.can_skip_application_steps or complete_applications.exists():
+            queryset = BootcampRun.objects.all().select_related("bootcamp")
+        else:
+            queryset = BootcampRun.objects.filter(
+                allows_skipped_steps=False
+            ).select_related("bootcamp")
```

Comment:
Finding a BootcampRunEnrollment for the user is sufficient in this case (we create one whenever an application is set to complete). This can also be cleaned up a little. I haven't tested this but something like this should work:

```suggestion
        user = self.request.user
        has_enrollments = BootcampRunEnrollment.objects.filter(user=user).exists()
        queryset = BootcampRun.objects.select_related("bootcamp")
        if not user.profile.can_skip_application_steps or not has_enrollments:
            queryset = queryset.filter(
                allows_skipped_steps=False
            )
```